Sources/Components/NativeAdvert/NativeAdvertDetailsContainer.swift

```diff
@@ -0,0 +1,154 @@
+        adRibbon.leadingAnchor.constraint(equalTo: adRibbonContainer.leadingAnchor, constant: .smallSpacing),
+        adRibbon.bottomAnchor.constraint(equalTo: adRibbonContainer.bottomAnchor, constant: -.verySmallSpacing),
+        adRibbon.trailingAnchor.constraint(equalTo: adRibbonContainer.trailingAnchor, constant: -.smallSpacing),
+
+        companyLabel.centerYAnchor.constraint(equalTo: adRibbonContainer.centerYAnchor),
+        companyLabel.leadingAnchor.constraint(equalTo: adRibbonContainer.trailingAnchor, constant: .smallSpacing),
+
+        titleLabel.leadingAnchor.constraint(equalTo: container.leadingAnchor),
+        titleLabel.trailingAnchor.constraint(equalTo: logoView.leadingAnchor, constant: -.mediumSpacing),
+        titleLabel.bottomAnchor.constraint(equalTo: container.bottomAnchor),
+
+        logoView.bottomAnchor.constraint(equalTo: bottomAnchor),
+        logoView.trailingAnchor.constraint(equalTo: trailingAnchor),
+    ]
+
+    private lazy var compactConstraints: [NSLayoutConstraint] = [
+        logoView.widthAnchor.constraint(equalToConstant: logoSizeCompact),
+        logoView.heightAnchor.constraint(equalToConstant: logoSizeCompact),
+
+        titleLabel.topAnchor.constraint(equalTo: adRibbonContainer.bottomAnchor, constant: .mediumSpacing),
+    ]
+
+    private lazy var regularConstraints: [NSLayoutConstraint] = [
+        logoView.widthAnchor.constraint(equalToConstant: logoSizeRegular),
+        logoView.heightAnchor.constraint(equalToConstant: logoSizeRegular),
+
+        titleLabel.topAnchor.constraint(equalTo: adRibbonContainer.bottomAnchor, constant: .mediumLargeSpacing),
+    ]
+
+    // MARK: - Init
+
+    override init(frame: CGRect) {
+        super.init(frame: .zero)
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        addSubview(container)
+
+        adRibbonContainer.addSubview(adRibbon)
+
+        container.addSubview(adRibbonContainer)
+        container.addSubview(companyLabel)
+        container.addSubview(titleLabel)
+        container.addSubview(logoView)
+
+        NSLayoutConstraint.activate(sharedConstraints)
+
+        setConstraints()
+        setFonts()
+    }
+
+    func configure(with model: NativeAdvertViewModel, andImageDelegate imageDelegate: NativeAdvertImageDelegate?) {
+        if let imageUrl = model.logoImageUrl {
+            imageDelegate?.nativeAdvertView(setImageWithURL: imageUrl, onImageView: logoView)
+        }
+
+        adRibbon.text = model.ribbonText
+        companyLabel.text = model.sponsoredBy
+        titleLabel.text = model.title
+    }
+
+    // MARK: - Overrides
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        setConstraints()
+        setFonts()
+    }
+
+    private func setConstraints() {
```

Comment:
Maybe you could rename the `setConstraints`  to something more clear about which constraints are affected, so `setupSizeClassConstraints` or something similar?